plugin/backend_lookup.go

```diff
@@ -403,4 +403,37 @@ func newAddress(s msg.Service, name string, ip net.IP, what uint16) dns.RR {
 	return &dns.AAAA{Hdr: hdr, AAAA: ip}
 }
 
+func checkZoneForRecord(b ServiceBackend, zone string, state *request.Request, opt Options) ([]msg.Service, error) {
+	var services []msg.Service
+	var err error
+	// if the zone name itself is queried we fake the query to search for a special entry
+	// this is equivalent to the NS search code
+	if state.Name() == zone {
+		old := state.QName()
+		state.Clear()
+		state.Req.Question[0].Name = "a.dns." + zone
```

Comment:
Can we not *also* use `ns.dns.` ? I.e we put different records here, and I can see people wanting to have MX and TXT as well.
Also for address record we need A and AAAA for starts which makes `a` a weird choice.